### DISCUSSION TITLE
🎉 Material web components 1.0 release

### DISCUSSION THREAD

**Discussion by asyncliz** [material-components/material-web#5004] [Announcements]
**tl;dr MWC is 1.0 and ready to use in production!**

After a long journey, Material Web has launched its 1.0 set of [web components](https://developer.mozilla.org/en-US/docs/Web/API/Web_components) using Google's latest [Material 3](https://m3.material.io/) design system!

Each component has been reviewed by Material 3 designers and accessibility engineers for use in production.

😱 🥳 👏 

Thank you to everyone who participated in our beta and provided feedback ❤️

**What's in 1.0?**

- ⚙️ 19 components, including buttons, checkboxes, chips, text fields, selects, and dialogs
- 🎨 CSS theming for dynamic color, fonts, and Material tokens
- ♿️ Accessibility for screen readers and high contrast
- 📖 Documentation and guides
- 📋 `<form>` element support
- 🏗️ Cross-framework support, including Lit, React, Vue, and Svelte
- 🤖 Basic [Lit SSR](https://lit.dev/docs/ssr/overview/) support
- 👏 Fewer breaking changes

Check out our [roadmap](https://github.com/material-components/material-web/blob/main/docs/roadmap.md) for what's coming next!

We're excited to help you build beautifully customized and accessible applications 🎉

The Material Web team

**Comment by miracleonyenma** [comment]
Finally 🎆🚀🙌🏾

**Comment by bivens-dev** [comment]
Congratulations! It’s been really nice to use so far, and it’s been exciting to see Lit really get used to deliver this. Historically, a lot of Google web stuff has felt incredibly tied to the internal toolchains and felt a bit distant from the rest of the web, this doesn’t give me those vibes at all.

**Comment by borisdelev** [comment]
Yey, what a great news. U awesome!
@asyncLiz I have a question about precompiled all-in-one CSS and JS bundles- is will be available by this way? (for easy quick start integration) 
Thanks!

**Reply by asyncliz** [reply]
Check out `@materal/web/all.js` and `@material/web/common.js`!

**Reply by ABugNamedBeetle** [reply]
one question! from where we will get the js files.

**Reply by asyncliz** [reply]
Use `npm install @material/web`. We also have a [quick start guide](https://github.com/material-components/material-web/blob/main/docs/quick-start.md)!

**Reply by borisdelev** [reply]
@asyncLiz did you consider CDN with precompiled bundle?

**Reply by treeder** [reply]
@borisdelev it's already available.  And even better, don't load all the components, just the ones you need on the page, see:

https://thingster.app/things/ctUhszB47Wb52JgHeMV9m

**Reply by e111077** [reply]
If you want to use a CDN and don't mind downloading all the components at once, we recommend `https://esm.run/@material/web/all.js` as esm.run will bundle and minify the bundle as well as split up dependencies like `lit`.

The only problem with esm.run is that it will not work if you intend to do individual component imports. If you want individual component imports and no bundling for CDN we recommend UNPKG with the `?module` flag e.g. `https://unpkg.com/@material/web/textfield/outlined-text-field.js?module`

**Reply by Krunal-AHAGURU** [reply]
How we can implement app designs in web

**Reply by dreamstar-enterprises** [reply]
I've read all the docs. I still can work out how to use this in my Angular 17 project. 

I do this, npm install @material/web, and then import a component, like this,

 import {MdOutlinedTextField} from '@material/web/textfield/outlined-text-field.js'

After that I try import MdOutlinedTextField, into the component, but cannot , as it complains that the import is for standalone components only

Then including this,  `<md-outlined-text-field label="Label" value="Value"></md-outlined-text-field>,` in the html, gives this:

Application bundle generation failed. [0.389 seconds]
⠸ Changes detected. Rebuilding...✘ [ERROR] TS-992012: Component imports must be standalone components, directives, pipes, or must be NgModules. [plugin angular-compiler]

    src/app/2. contract-module/edit-tabs/personal-information-edit/personal-information-edit.component.ts:27:120:
      27 │ ...t, FieldEntryComponent, ReactiveFormsModule, MdOutlinedTextField],

Sorry, I'm a beginner : (

**Reply by asyncliz** [reply]
You can post a [new Q&A discussion](https://github.com/material-components/material-web/discussions/new?category=q-a) to get better help visibility!

**Reply by dreamstar-enterprises** [reply]
Done!

**Comment by Nick-DL** [comment]
Now it can finally be used for my website!

**Comment by mreiche** [comment]
What a great documentation. Thanks a lot guys!

**Comment by sondreb** [comment]
Great work, congratulations! Got it working with Angular fairly easy. So very nice customization of themes and colors, love it. Won't migrate fully from Angular Material yet, but happy this is coming along.

**Reply by Hackmancoltaire** [reply]
Any advice for getting this working with Angular?

**Comment by gilbertoesp** [comment]
Thank you so much. I am going for a job being a web developer and designing is not very much my field of expertise. With this kind of material I can focus in what I like. Building solutions

**Comment by maceip** [comment]
super excited! just to check are touch events supported?

**Reply by e111077** [reply]
they should be. Please file GitHub issues if you run into touch issues

**Comment by BLM3826** [comment]
Congratulations on the launch! I am curious, will these components have any color compatibility with the theme of the Chrome browser in any way? It would be nice to have an option like that, given the fact that in Android the Material You apps (made by Google) are aligned with the color theme of the phone.

**Reply by mon-jai** [reply]
I don't think there is a web API for that (yet).

**Reply by e111077** [reply]
The Chrome OS team uses these components and are investigating a possible solution for this specifically for installed web apps. Depending on their findings and possible successes, hopefully we can investigate opening this up to Chrome in general

**Comment by Char2sGu** [comment]
Congrats! We finally have an official component library to implement MD3 in our apps!

**Reply by Char2sGu** [reply]
And also hope Angular Material is gonna catch up with this update soon!

**Comment by JosefJezek** [comment]
here is a build-less DEMO https://jsbin.com/sojocuf/1/edit?html,output

thank you for these awesome components 👍🏻 🎉

**Comment by JosefJezek** [comment]
SSR DEMO https://stackblitz.com/edit/github-uiraud?file=src%2Fpages%2Findex.astro

**Comment by QZ-harkle** [comment]
yes sir

**Comment by Caritoux** [comment]
I was waiting for this! As a designer, I'm a big fan of Material Design, and I want to bring alive my project with Material Web, unfortunately, many components are not yet ready like Navigation 💔

**Comment by daviavmello** [comment]
Great project, but I thought this would be incorporated into Angular Material first?

**Reply by e111077** [reply]
Angular Material is currently working with our sister team that builds Material for an internal Google framework to share infrastructure. That team _also_ builds a designer-reviewed reference implementation just like this library. Though, these components [should work in Angular today](https://stackblitz.com/edit/stackblitz-starters-ofrtvb?description=Starter%20project%20for%20Node.js,%20a%20JavaScript%20runtime%20built%20on%20Chrome%27s%20V8%20JavaScript%20engine&file=package.json,src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.component.html&title=node.new%20Starter).

We intend to create wrappers for our components in React as to help smooth out the interaction in that framework, but nobody on our team has touched Angular in quite a few years. If you run into any issues / weirdness using this library in Angular, please file issues on us as we need to form an idea on what is missing and what we can do to plug those development inconsistencies to make developing with these components more idiomatic in Angular.

**Reply by daviavmello** [reply]
Thank you for your response, Elliot. Should we expect the Angular Material website to have a big overhaul in the documentation and look/feel, too? Just for reference, it would be appealing to see an Angular Material as strong as MUI is for React.

**Reply by e111077** [reply]
I would like to reiterate that this project is not Angular Material and currently not the basis for Angular Material components. I unfortunately do not have any insight into the future of Angular Material's documentation site.

What we're hoping to do in this project is to build a super-fast, stable, maintainable, accessible, cross-platform, reference implementation of Material Design for use at Google as well as open source projects such as React, Lit, Vue, Svelte, SolidJS, etc. The Angular team is building Angular Material specifically for Angular users. In the meantime we don't mind accepting Angular issues to bridge the gap until NG Material 3 is ready, but know that their angular-focused library is in development and that we cannot ensure or promise API compatibility with the future Angular Material 3 library.

**Reply by daviavmello** [reply]
Thank you, Elliott! I appreciate the info.

**Comment by Guillermo-prog** [comment]
What's the difference between this new repository release and this older one? https://github.com/material-components/material-components-web

**Reply by asyncliz** [reply]
The older one (we call it MDC) originally had a goal of being a framework-agnostic implementation for Material. In reality, the patterns we used were not as scalable as we wanted.

We created MWC to replace it as a stronger framework-agnostic implementation using web components. The former MDC library is evolving to focus on supporting Angular Material.

Both Material projects have different goals and needs, but are built from the same core foundation of Material tokens. We recommend using Angular Material for Angular projects since it offers more benefits due to its tight Angular integration, and MWC for multi-framework or non-Angular projects.

**Reply by mon-jai** [reply]
The old one implements Material 2. This repo implements the latest Material Design 3 guideline.

**Comment by vaynevayne** [comment]
hi, what are  difference wetween mui and this ?  https://mui.com/material-ui/

**Reply by asyncliz** [reply]
MUI is a third-party library that provides Material 2 components. This library is from Google and provides the latest Material 3 components.

You can also mix and match them if you want to use a lot of MUI's great layout and app building utilities, or components that don't have an implementation here yet :)

**Comment by zyunx** [comment]
Congratulations!!!

**Comment by bdabak** [comment]
Very well done guys! It is finally a genuine web release. I am looking forward to seeing all the upcoming missing components 🤩

**Comment by MarcelGeo** [comment]
Good job ✅✅🎉🎉

**Comment by for-the-zero** [comment]
I think it's imperfect, needs to be improved

**Comment by QZ-harkle** [comment]
Congrats on the improvement
